### PR TITLE
Fix crash when running SA1623 code fix

### LINKED ISSUE #1943: SA1623 code fix throws exception for empty summary
The code fix for SA1623 throws an exception when attempting to fix the following code:

```csharp
public class ClassName
{
    /// <summary></summary>
    public int Property
    {
        get;
    }
}
```

### PR DESCRIPTION
**Pull request by NikolayIT**
Fix #1943

### CODE REVIEW

**Comment by NikolayIT** [comment]
We have to discuss what should the default text of the code fix for SA1623 be.

Currently it is always `<summary>Gets the value</summary>`.

I am having a suggestion to replace it with `<summary>Gets PropertyName</summary>`.

What to you think?

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs [hunk 1]
```diff
@@ -152,6 +152,39 @@ private class PrivateTestClass
...
+    /// <summary></summary>
+    public int Property
+    {
+        get;
+    }
+}";
+
+            var fixedTestCode = @"public class ClassName
+{
+    /// <summary>Gets the value</summary>
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs]: :exclamation: This suppresses the warning but does not actually fix the documentation problem. For other similar situations, we intentionally do not provide a code fix, and I think we should continue to follow that model here.

Comment (NikolayIT) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs]: OK, tomorrow I will remove the default summary and will leave it empty.

**Comment by NikolayIT** [comment]
@sharwell, done.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs [hunk 2]
```diff
@@ -152,6 +152,33 @@ private class PrivateTestClass
...
+}";
+
+            var expected = this.CSharpDiagnostic(PropertySummaryDocumentationAnalyzer.SA1623Descriptor).WithLocation(4, 16).WithArguments("Gets");
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+
+            // Intentionally do not provide a code fix: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/1957#discussion_r47703520
+            //// await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+
+            await this.VerifyCSharpFixAsync(testCode, testCode).ConfigureAwait(false);
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs]: :thought_balloon: This should probably use:
```csharp
var offeredFixes = await this.GetOfferedCSharpFixesAsync(testCode).ConfigureAwait(false);
Assert.Empty(offeredFixes);
```

**Comment by vweijsters** [comment]
Currently a code fix is offered that will do nothing. I would prefer to see it changed so that the code fix is not offered at all. It will be more clear for the user that the diagnostic cannot be fixed by us.

**Comment by sharwell** [comment]
> Currently a code fix is offered that will do nothing. I would prefer to see it changed so that the code fix is not offered at all. It will be more clear for the user that the diagnostic cannot be fixed by us.

We could probably do this efficiently by adding a property to the reported diagnostic in the case where the summary is empty. That way the code fix can simply skip over diagnostics that have the property without performing any additional analysis.

**Comment by NikolayIT** [comment]
So what should I do in order to match your CR? :smile:

**Comment by vweijsters** [comment]
@NikolayIT Have a look at the SA1003 analyzer and code fix. They implement a similar mechanism. 

In this case you can simplify stuff a bit. The presence of the property in the set can be used to check if the code fix should be offered or not.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs (hunk 2 above)
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1623UnitTests.cs]: Interesting, I didn't even know we had that. :smile:

**Comment by vweijsters** [comment]
:+1:

**Comment by sharwell** [comment]
:+1:

**Pull request merged**